src/Utilities/Compiler/Lightup/LightupHelpers.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities.Lightup
+{
+    internal static class LightupHelpers
+    {
+        internal static Func<TSyntax, TProperty> CreateSyntaxPropertyAccessor<TSyntax, TProperty>(Type type, string propertyName)
+            where TSyntax : SyntaxNode
+            => CreatePropertyAccessor<TSyntax, TProperty>(type, "syntax", propertyName);
+
+        internal static Func<TSymbol, TProperty> CreateSymbolPropertyAccessor<TSymbol, TProperty>(Type type, string propertyName)
+            where TSymbol : ISymbol
+            => CreatePropertyAccessor<TSymbol, TProperty>(type, "symbol", propertyName);
+
+        private static Func<T, TProperty> CreatePropertyAccessor<T, TProperty>(Type type, string parameterName, string propertyName)
+        {
+            static TProperty FallbackAccessor(T instance)
```

Comment:
Can `TProperty` be constrained to `struct`? then we don't need to suppress on return here

Reply:
nit: Also, I thought the IDE adopted the convention of putting local functions last. The compiler also uses this convention, as it helps keep the body of the method front and center.